commercial-schedules/src/main/java/org/dcsa/conformance/standards/cs/checks/CsChecks.java

```diff
@@ -0,0 +1,393 @@
+
+    // Compare the dates
+    if (dateType.equals("startDate") && !dateTimeAsDate.isAfter(date)) {
+      return String.format("The %s date should be after the %s start date", operation, operation);
+    } else if (dateType.equals("endDate") && !dateTimeAsDate.isBefore(date)) {
+      return String.format("The %s date should be before the %s end date", operation, operation);
+    }
+    return "";
+  }
+
+  private static JsonContentCheck validateUSRForVs(
+      Supplier<SuppliedScenarioParameters> sspSupplier) {
+    return JsonAttribute.customValidator(
+        "Validate USR available in the response",
+        body -> {
+          var issues = new LinkedHashSet<String>();
+          for (JsonNode vs : body) {
+            var csp = sspSupplier.get().getMap();
+            if (csp.containsKey(UNIVERSAL_SERVICE_REFERENCE)) {
+              if (vs.at("/universalServiceReference").isMissingNode()) {
+                issues.add(
+                    "The 'universalServiceReference' needs to be provided in the response if it is given in the filter.");
+              }
+            }
+          }
+          return issues;
+        });
+  }
+
+  private static JsonContentCheck validateDateForPs(
+      Supplier<SuppliedScenarioParameters> sspSupplier) {
+    return JsonAttribute.customValidator(
+        "Validate date in the response",
+        body -> {
+          var issues = new LinkedHashSet<String>();
+          for (JsonNode schedule : body) {
+            schedule
+                .at("/vesselSchedules")
+                .forEach(
+                    vesselSchedule -> {
+                      vesselSchedule
+                          .at("/timestamps")
+                          .forEach(
+                              timestamp -> {
+                                JsonNode eventDateTime = timestamp.at("/eventDateTime");
+                                JsonNode eventClassifierCode = timestamp.at("/eventClassifierCode");
+                                if (eventClassifierCode.asText().equals("EST")) {
+                                  if (isBeforeTheDate(
+                                      eventDateTime.asText(),
+                                      sspSupplier.get().getMap().get(DATE))) {
+                                    issues.add(
+                                        "The estimated arrival or departure dates should be on or after the date provided");
+                                  }
+                                } else if (eventClassifierCode.asText().equals("PLN")) {
+                                  if (isBeforeTheDate(
+                                      eventDateTime.asText(),
+                                      sspSupplier.get().getMap().get(DATE))) {
+                                    issues.add(
+                                        "The estimated arrival or departure dates should be on or after the date provided");
+                                  }
```

Comment:
My IDE gives a warning: This branch's code block is the same as the block for the branch on line 344.
Can you address the duplication?